app/utils/data.py

```diff
@@ -0,0 +1,149 @@
+# Copyright (C) 2020-2023, Pyronear.
+
+# This program is licensed under the Apache License 2.0.
+# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0> for full license details.
+
+import ast
+import json
+from io import StringIO
+from pathlib import Path
+from typing import List
+
+import pandas as pd
+
+
+def read_stored_DataFrame(data):
+    """
+    Reads a JSON-formatted string representing a pandas DataFrame stored in a dcc.Store.
+
+    Args:
+        data (str): A JSON-formatted string representing the stored DataFrame.
+
+    Returns:
+        tuple: A tuple containing the DataFrame and a boolean indicating whether data has been loaded.
+    """
+    # if "false" in data:
+    #     return pd.DataFrame().to_json(orient="split"), False
+    data_dict = json.loads(data)
+
+    # Check if 'data' is empty or if 'columns' is empty
+    if not len(data_dict["data"]):
+        # If either is empty, create an empty DataFrame
+        return pd.DataFrame().to_json(orient="split"), data_dict["data_loaded"]
+    else:
+        # Otherwise, read the JSON data into a DataFrame
+        return (
+            pd.read_json(StringIO(data_dict["data"]), orient="split"),
+            data_dict["data_loaded"],
+        )
+
+
+def process_bbox(input_str):
+    """
+    Processes the bounding box information from a xyxy string input to a xywh list of integer coordinates.
+
+    Args:
+        input_str (str): A string representing the bounding box coordinates.
```

Comment:
we send the precision now just in case we update the display strategy (having a cursor base on confidence for exemple) but bbox are already filtered on confidence score in pyro-egine. I will rework this part soon

Reply:
this is old code